helix-core/src/test/resources/TestRecoveryLoadBalance.MasterSlave.json

```diff
@@ -253,5 +253,115 @@
         }
       }
     }
+  },
+  {
+    "statemodel": "MasterSlave",
+    "errorOrRecoveryPartitionThresholdForLoadBalance": 0,
+    "loadBalanceThrottle": "0",
+    "minActiveReplica": "2",
+    "inputs": {
+      "resource_0_0": {
+        "currentStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE",
+          "localhost_3": "OFFLINE"
+        },
+        "bestPossibleStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "SLAVE",
+          "localhost_3": "DROPPED"
+        },
+        "expectedStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE",
+          "localhost_3": "OFFLINE"
+        }
+      },
+      "resource_0_1": {
+        "currentStates": {
+          "localhost_0": "SLAVE",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE"
+        },
+        "bestPossibleStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "SLAVE"
+        },
+        "expectedStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "SLAVE"
+        }
+      }
+    }
+  },
+  {
+    "statemodel": "MasterSlave",
+    "errorOrRecoveryPartitionThresholdForLoadBalance": 0,
+    "loadBalanceThrottle": "2",
```

Comment:
Optional, but I think it would be better to also add a case where `loadBalanceThrottle` is 1?

Reply:
1 will not work. Because partition 0 and 1 are same partition priority level. Sometimes it could be 0 throttled or 1 throttled. I would like to test the partition 0 with all downwarding state transition but partition 1 with normal state transitions.

Reply:
Synced up with Hunter offline. It is about internal state transition priority.